R/backend-sqlite.R

```diff
@@ -129,7 +129,11 @@ sql_query_wrap.SQLiteConnection <- function(con, from, name = unique_subquery_na
     if (is.null(name)) {
       build_sql(sql_indent_subquery(from, con, lvl), con = con)
     } else {
-      build_sql(sql_indent_subquery(from, con, lvl), " AS ", ident(name), con = con)
+      ident_name <- name %||% unique_subquery_name()
+      if (!is.ident(ident_name)) {
```

Comment:
Since this seems to be a repeated pattern, might be worth extracting it out?

```R
as_subquery_name <- function(x) {
  if (is.ident(x)) {
    x
  } else if (is.null(x) {
    unique_subquery_name()
  } else {
    ident(x)
  }
}
```